File: hashes.js
let person = {
    name: "John",
    age: 30,
    city: "New York"
  }
//   Object.keys()
console.log(Object.keys(person))
// Object.values()
console.log(Object.values(person))
// .length
console.log(Object.keys(person).length)
// .notation
console.log(person.name)
console.log(person.gender = "unknown")
// the in method to check if a key exists
console.log("name" in person)
console.log("height" in person)
// to check if a value exists in an object
console.log(Object.values(person).includes("New York"))
console.log(Object.values(person).includes("New Yorker"))
// merging two objects using the spread operator
let person1 = {
    name: "John",
    age: 30,
    city: "New York"
  }
  let person2 = {
    name: "Jane",
    gender: "Female"
  }
  person = {...person1, ...person2}
  console.log(person)

File: hashes.rb
person = {
  name: "John",
  age: 30,
  city: "New York"
}
# .keys
puts person.keys()
# .values
puts person.values()
# .length / .size
puts person.length()
# .fetch()
puts person.fetch(:name)
puts person.fetch(:gender, "Unknown") 
#  .key?
puts person.key?(:name)
puts person.key?(:gender)
# .has_value? or .value?
puts person.has_value?("John") 
puts person.value?("Jane") 
# .merge()
person1 = {
    name: "John",
    age: 30,
    city: "New York"
  }
  person2 = {
    name: "Jane",
    gender: "Female"
  }
  person = person1.merge(person2)
  puts person
  